lib/spec_file_generator.js

```diff
@@ -19,10 +19,17 @@ function sorted(items) {
 }
 
 function dependenciesToSources(deps) {
+  deps.sort((a, b) => {
+    if (a[0] === b[0]) {
+      return a[1].localeCompare(b[1]);
+    } else {
+      return a[0].localeCompare(b[0]);
+    }
+  });
   const sources = deps.map((dependency, index) => {
```

Comment:
This function is now very similar to `dependenciesToProvides`. It looks like that method can actually be used there (probably with a better name):
```suggestion
   const sources = dependenciesToProvides(deps).map((dependency, index) => {
```